### PR TITLE
tenant: reliable and consistent add/delete operations (fixes #1158)

### LINKED ISSUE #1158: Operations `add` and `delete` via `keylime_tenant` CLI need to produce consistent results
Is your issue a feature request? If so, please raise it as an [enhancement](https://github.com/keylime/enhancements)

# Environment
* OS / version: Fedora Core 35
* Processor architecture: x86_64
* TPM Manufacturer: swtpm
* Keylime version: 6.5.1

# Description 
When running the command `keylime_tenant -c add` in a large cluster (~ 1,500 `agents`), with a relatively long allowlist (~ 80,000 entries) and an external `mysql` database, it can be observed  that the API returns the correct response code, the command end with zero exit code, and yet, there are occasions when the actual entry pertaining the `agent`  has not yet been written to the database, causing a subsequent immediate `keylime_tenant -c cvstatus` execution to falsely report the `agent` as "not yet added to a `verifier`".

## Expected behavior vs. actual behavior 
For both `keylime_tenant -c add` and ` keylime_tenant -c delete` (or `keylime_tenant -c update`), the CLI should return only after querying the actual status of the `agent` from the `verifier` via a new API call, and then ensuring that desired state has been reached. Currently, `keylime_tenant -c delete` already does it via a crude "for" loop, but this should be properly implemented for all the aforementioned commands, using a backoff algorithm.

## Steps to reproduce problem
So far, this has only detectable under the specific conditions listed at the description

# Relevant logs
NA

### PR DESCRIPTION
**Pull request by maugustosilva**
As discussed in issue #1158, there are real-world scenarios where "add" and/or "delete" operations via `keylime_tenant` CLI causes it to return before the data was written by `verifier` on the database. This causes inconsistent "status" (also via `keylime_tenant`) for a given `agent`, in case the CLI invoked within a small time window after "add"/"delete".

Given the fact the code for `keylime_tenant` was somewhat messy (e.g., a fixed 12-step loop with 0.4 seconds between attempts on "delete"), this new code uses the already existing "retry_interval", "exponential_backoff", and "max_retries" attributes to implement an ordered "status" polling on the `verifier` right after an "add" or "delete" operation, only returning when the desired state and the actual state effectivelly match (or produce an explicit error message in case it does not happen after multiple retries)

Signed-off-by: Marcio Silva <marcio.a.silva@ibm.com>

### CODE REVIEW

**Comment by THS-on** [comment]
The general code LGTM. I think "appear" is not ideal for the messages because it suggests that stuff can get lost somewhere. I'm for writing it in a more active form e.g.: "Agent is still not deleted from the verifier's database".

Some of the status messages changes, so that is the reason why the e2e test fail. @kkaarreell what is the best way to change those?

We should be consistent with the `sys.exit()` calls, because some of the `do_*` functions return a dict and others just exit.
I think we should move to a model of either returning data or throwing an error exception that then can be handled, but that is for another PR.

File: keylime/tenant.py
Comment (aplanas) [inline keylime/tenant.py:108]: Are those new options? Should we do a config migration? Add some defaults?

Comment (THS-on) [reply keylime/tenant.py:108]: There were already there, so no migration or default options needed: https://github.com/keylime/keylime/blob/9dae00e519ba5f37cc7b8bfbebdb2ad6953f03bc/templates/2.0/tenant.j2#L78

File: keylime/tenant.py
Comment (mpeters) [inline keylime/tenant.py:598]: I agree with Thore about using "appear", it's too wishy-washy.

**Review by mpeters — CHANGES_REQUESTED** [review]
Except for the wording on the user messages, this LGTM

File: keylime/tenant.py
Comment (maugustosilva) [reply keylime/tenant.py]: fixed :-)

File: keylime/tenant.py
Comment (mpeters) [inline keylime/tenant.py:598]: I still think this isn't direct enough. Maybe something like "Agent ID %s still not added to Verifier after %d tries", then repeat for the rest. We don't need to tell the user we keep track of it with a specific status value or how we track that internally, just the logical action of was it added or not.

**Review by mpeters — CHANGES_REQUESTED** [review]

File: keylime/tenant.py
Comment (maugustosilva) [reply keylime/tenant.py]: Fixed now

**Review by THS-on — APPROVED** [review]
Before merging the tests should be still adjusted to the new messages, but otherwise LGTM. Thanks for adding that because during testing I also ran into those issues sometimes. 

Also it should also fix: https://github.com/keylime/keylime/issues/960

**Comment by maugustosilva** [comment]
> Before merging the tests should be still adjusted to the new messages, but otherwise LGTM. Thanks for adding that because during testing I also ran into those issues sometimes.
> 
> Also it should also fix: #960

I have fixed the messages. Now I see the `rawhide` error that affects everything (@kkaarreell sent an email to devs, waiting for an answer) and an unrelated error on `centos-stram-9-x86_64

**Comment by THS-on** [comment]
> I have fixed the messages. 

Yes I've only saw the old test run when I did the review. The other failing tests have nothing to do with this, so it is ready to merge. (I cannot merge it because @mpeters still needs to approve this, because he requested some changes)

**Review by mpeters — APPROVED** [review]

**Pull request merged**
